### PR TITLE
Add --dart-define option support to build aar command

### LINKED ISSUE #71827: [flutter_tools]Add --dart-define option support to build aar command.
## Use case
Now as we can use --dart-define to provide custom constant at build time with `flutter run`, `flutter build apk`, `flutter build ios`... build commands, except `flutter build aar`.
image

## Proposal
Add this support to make `flutter build aar` more conveniently

### PR DESCRIPTION
**Pull request by hyiso**
## Description
Add --dart-define option support to `flutter build aar` command

## Related Issues
https://github.com/flutter/flutter/issues/71827

### CODE REVIEW

**Comment by hyiso** [comment]
@googlebot I signed it!

**Review by jmagman — CHANGES_REQUESTED** [review]
This needs a test.

See https://github.com/flutter/flutter/blob/6a2bc2617e0b0197f33e00ce804f81b4b93474c0/packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart#L161

**Comment by hyiso** [comment]
Sorry for late, I will add some test code for this PR on this weekend.

File: packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart [hunk 1]
```diff
@@ -196,6 +197,8 @@ void main() {
       expect(buildInfo.flavor, 'free');
       expect(buildInfo.splitDebugInfoPath, '/project-name/v1.2.3/');
       expect(buildInfo.dartObfuscation, isTrue);
+      expect(buildInfo.dartDefines.isNotEmpty, isTrue);
```
Comment (jmagman) [inline packages/flutter_tools/test/commands.shard/permeable/build_aar_test.dart:200]: Nit: this check isn't necessary with the `contains` check below.

**Review by jmagman — COMMENTED** [review]
Looks good other than the nit about removing the extra `isNotEmpty`.

**Review by jmagman — APPROVED** [review]
Looks good, thanks for the contribution!

**Pull request merged**